node.go

```diff
@@ -2,57 +2,67 @@ package main
 
 import (
 	"context"
-	"crypto/rand"
-	"crypto/subtle"
 
 	"github.com/CCI-MOC/obmd/internal/driver"
+	"github.com/CCI-MOC/obmd/token"
 )
 
 // Information about a node
 type Node struct {
 	ConnInfo     []byte             // Connection info for this node's OBM.
 	ObmCancel    context.CancelFunc // stop the OBM
 	OBM          driver.OBM         // OBM for this node.
-	CurrentToken Token              // Token for regular user operations.
+	CurrentToken token.Token        // Token for regular user operations.
 }
 
 // Returns a new node with the given driver information, with no valid token.
 func NewNode(d driver.Driver, info []byte) (*Node, error) {
+	tok, err := token.New()
+	if err != nil {
+		return nil, err
+	}
+
 	obm, err := d.GetOBM(info)
 	if err != nil {
 		return nil, err
 	}
 	ret := &Node{
-		OBM:      obm,
-		ConnInfo: info,
+		OBM:          obm,
+		ConnInfo:     info,
+		CurrentToken: tok,
 	}
-	copy(ret.CurrentToken[:], noToken[:])
 	return ret, nil
 }
 
 // Generate a new token, invaidating the old one if any, and disconnecting
 // clients using it. If an error occurs, the state of the node/token will
 // be unchanged.
-func (n *Node) NewToken() (Token, error) {
-	var token Token
-	_, err := rand.Read(token[:])
+func (n *Node) NewToken() (token.Token, error) {
```

Comment:
Why don't we directly call n.OBM.dropconsole() instead of calling ClearToken? since we are generating a new token here anyway.

Reply:
This way the logic for invalidating the token is encapsulated by ClearToken. But NewToken() isn't really doing anything else, so maybe now that we don't have this noToken construct we should just combine them. Don't feel strongly.